src/firefly/js/templates/lightcurve/LcPeriodogram.jsx

```diff
@@ -505,13 +509,12 @@ function periodogramSuccess(popupId, hideDropDown = false) {
             /*sortInfo: sortInfoString('Power', false)*/
         }, {tbl_id: LC.PERIODOGRAM_TABLE, inclCols : '"Period", "Power"'});
 
-
         if (tReq !== null) {
             dispatchTableSearch(tReq, {removable: true, tbl_group: LC.PERIODOGRAM_GROUP});
             const dispatchParams= {
                 viewerId: LC.PERIODOGRAM_GROUP,
                 groupId: LC.PERIODOGRAM_TABLE,
-                chartId: LC.PERIODOGRAM_TABLE,
+                chartId: LC.PERIODOGRAM_TABLE + '_' + periodogramChartCount++,
```

Comment:
Are charts getting cleaned up if you do this? You should check with @loitly I am concern with will be a memory leak.

Reply:
Yes good point. I'll check again. From what I can tell with a quick look (by doing a `getChartData`), it does seem like the previous charts are getting cleared though.

Reply:
Yes @robyww, everytime we calculate or recalculate periodogram, we do a `dispatchTableSearch` and the `tbl_id` for this search always remains the same (`LC.PERIODOGRAM_TABLE`) so even though the `chartId` changes, the previous chart is cleared as the table search will internally do a `dispatchTableRemove` on this `tbl_id`.

Reply:
I think there is something like `dispatchChartRemove` (I have not checked). That might need to be called as well with the previous id. If so it could be setup to be called in the cleanup useEffect return function.